### PR TITLE
[graphql/rpc] chore/easy: add TODO comments to callout we're moving away from sdk usage

### PR DESCRIPTION
**Pull request by oxade**
## Description 

Make it more explicit via comments that we're moving away from depending on RPC 1.0 SDK so we can migrate and cleanup as needed.

## Test Plan 

N/A

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/types/transaction_block.rs [hunk 1]
```diff
@@ -59,6 +59,7 @@ impl TransactionBlock {
             return Ok(None);
         }
         let gcs = self.effects.as_ref().unwrap().gas_effects.gcs;
+        // TODO: implement DB counterpart without using Sui SDK client
         let data_provider = ctx.data_provider();
         let system_state = data_provider.get_latest_sui_system_state().await?;
         let protocol_configs = data_provider.fetch_protocol_config(None).await?;
```
Comment (oxade) [inline crates/sui-graphql-rpc/src/types/transaction_block.rs:65]: @wlmyng I wouldn't consider transaction block done while these are here.
Is the DB impl feasible yet?

Comment (wlmyng) [reply crates/sui-graphql-rpc/src/types/transaction_block.rs:65]: @stefan-mysten is working on this (transactionBlock.expiration) field, should be doable as expiration is actually a field returned from StoredTransaction

**Review by wlmyng — APPROVED** [review]

File: crates/sui-graphql-rpc/src/types/transaction_block.rs (hunk 1 above)
Comment (stefan-mysten) [reply crates/sui-graphql-rpc/src/types/transaction_block.rs:65]: Yes, a PR should come in shortly once I figure some really weird errors.

Comment (stefan-mysten) [reply crates/sui-graphql-rpc/src/types/transaction_block.rs:65]: #14147

**Pull request merged**